frontend/utils/tokenUtils.ts

```diff
@@ -25,25 +25,34 @@ export const lspInterfaceShortNames = {
   [INTERFACE_IDS.LSP8IdentifiableDigitalAsset]: 'LSP8',
 };
 
+function computeSelector(signature: string): string {
+  return ethers.utils
+    .keccak256(ethers.utils.toUtf8Bytes(signature))
+    .slice(0, 10); // First 4 bytes (8 characters + '0x')
+}
+
+function supportsFunction(bytecode: string, selector: string): boolean {
+  return bytecode.includes(selector.slice(2)); // Remove '0x' when searching in bytecode
+}
+
+const lsp7TransferSelector = computeSelector(
+  'transfer(address,address,uint256,bool,bytes)'
+);
+const lsp8TransferSelector = computeSelector(
+  'transfer(address,address,bytes32,bool,bytes)'
+);
+
 export const detectLSP = async (
   assetAddress: string
 ): Promise<string | null> => {
   // fetch digital asset interface details
   try {
-    const lspAsset = new ERC725(
-      lsp4Schema as ERC725JSONSchema[],
-      assetAddress,
-      getLuksoProvider()
-    );
-    const isLSP7 = await lspAsset.supportsInterface(
-      INTERFACE_IDS.LSP7DigitalAsset
-    );
+    const bytecode = await getProvider().getCode(assetAddress);
+    const isLSP7 = supportsFunction(bytecode, lsp7TransferSelector);
     if (isLSP7) {
       return INTERFACE_IDS.LSP7DigitalAsset;
     }
-    const isLSP8 = await lspAsset.supportsInterface(
-      INTERFACE_IDS.LSP8IdentifiableDigitalAsset
-    );
+    const isLSP8 = supportsFunction(bytecode, lsp8TransferSelector);
```

Comment:
@democrazyfi can we please separate out unrecognized from lsp7 vs lsp8, im just concerned `supportsFunction` is a much weaker check than `supportsInterface`. `supportsInterface` gives us better guarantees for all other assumptions around data structure and functionality

Reply:
How do you mean separate unrecongized?

I think checking the bytecode is safer because it gurantees that a function with that signture exists

But as this docs mentions https://docs.lukso.tech/contracts/interface-ids/ 
```
Interface IDs are not the most secure way to ensure that a contract implements a specific set of functions, as they are manually set and can be set to any value.
```

Reply:
How do you mean separate unrecongized?

I think checking the bytecode is safer because it gurantees that a function with that signture exists

But as this docs mentions https://docs.lukso.tech/contracts/interface-ids/ 
`
Interface IDs are not the most secure way to ensure that a contract implements a specific set of functions, as they are manually set and can be set to any value.
`

Reply:
I mean have a separate designation. unrecognized LSP7 and unrecognized LSP8. interface isnt perfect but it tells more than transfer function about the asset overall. bytecode of transfer guarantees revival but not structure of other aspects